lib/ProviderAws.js

```diff
@@ -217,6 +217,12 @@ module.exports = function(S) {
         .then(() => this.addProfileCredentials(credentials, 'AWS_' + stage))                            // allow for *stage specific* Amazon standard prefix based profile declaration
         .then(() => this.addProfileCredentials(credentials, 'SERVERLESS_ADMIN_AWS_' + stage))           // allow for *stage specific* Serverless standard prefix based profile declaration
         .then(() => {
+          if (!credentials.accessKeyId || !credentials.secretAccessKey) {
+            // fall back to attempting EC2 Metadata Service (this is a fallback
+            // because it involves an http call that may time out on the order
+            // of seconds in case this is not an EC2 instance)
+            this.addEc2MetadataCredentials(credentials, 'AWS')
```

Comment:
This would do something like:

``` 
var AWS = require('aws-sdk');
AWS.config.getCredentials(function(err) {
  if (err) console.log(err.stack); // credentials not loaded
  else { 
    console.log("Access Key:", AWS.config.credentials.accessKeyId);
    // pass AWS.config.credentials properties to your SimpleDB code
  }
});
```

from https://forums.aws.amazon.com/thread.jspa?threadID=164567